content/08-mongodb/09-mongodb-indexes.mdx

```diff
@@ -0,0 +1,260 @@
+
+```
+[
+   {
+      "v" : 2,
+      "key" : {
+         "country" : 1
+      },
+      "name" : "country"
+   }
+ ]
+```
+Index information includes the keys and options used to create the index.
+
+<PrismaOutlinks>
+
+If you're interested in using full text indexes with MongoDB, Prisma has the [`fullTextIndex` preview feature](https://www.prisma.io/docs/concepts/components/prisma-schema/indexes#full-text-indexes-mysql-and-mongodb) which allows you to easily migrate full text indexes into a Prisma schema for type safety and protection against validation errors.
+</PrismaOutlinks>  
+
+## How to understand index performance
+Now with the ability to create and check what indexes exist on your collection, you will want to see if your indexes are performing the way you expect.
+
+To begin the example, we will work with the `sample_mflix` database and the `comments` collection which has ~50.3k documents. This is a sample collecton provided by MongoDB simulating a datastore of movie and tv comments. 
+
+To understand the performance of an index, we will want to first run a query that does not have an index. The following query will return all 273 documents in the collection that are comments made by `Ramsay Bolton`:
+
+```
+db.comments.find( { "name" : "Ramsay Bolton" } )
+```
+Now, if we append [MongoDB's explain plan](https://docs.mongodb.com/manual/reference/explain-results/) to the query, we will see the performance of this query.
+
+```
+db.comments.find( { "name" : "Ramsay Bolton" } ).explain("executionStats")
+```
+This results in the following:
+
+```
+{
+  queryPlanner: {
+    plannerVersion: 1,
+    namespace: 'sample_mflix.comments',
+    indexFilterSet: false,
+    parsedQuery: { name: { '$eq': 'Ramsay Bolton' } },
+    winningPlan: {
+      stage: 'COLLSCAN',
+      filter: { name: { '$eq': 'Ramsay Bolton' } },
+      direction: 'forward'
+    },
+    rejectedPlans: []
+  },
+  executionStats: {
+    executionSuccess: true,
+    nReturned: 273,
+    executionTimeMillis: 23,
+    totalKeysExamined: 0,
+    totalDocsExamined: 50303,
+    executionStages: {
+      stage: 'COLLSCAN',
+      filter: { name: { '$eq': 'Ramsay Bolton' } },
+      nReturned: 273,
+      executionTimeMillisEstimate: 6,
+      works: 50305,
+      advanced: 273,
+      needTime: 50031,
+      needYield: 0,
+      saveState: 50,
+      restoreState: 50,
+      isEOF: 1,
+      direction: 'forward',
+      docsExamined: 50303
+    }
+  }
+}
+```
+There are several key results to focus on in this output. First, we can see in the `winningPlan` that the `stage` for this query is a `COLLSCAN`. This means that a collection scan occurred to commplete this query with a `totalDocsExamined` of 50,303 and `executionTimeMillis` of 23 milliseconds. The query had to examine every document in the collection and took 23ms even though `nReturned` was only 273 documents. While 23ms may not sound like much, this can get much longer for a collection housing one million documents.
```

Comment:
```suggestion
There are several key results to focus on in this output. First, we can see in the `winningPlan` that the `stage` for this query is a `COLLSCAN`. This means that a collection scan occurred to complete this query with a `totalDocsExamined` of 50,303 and `executionTimeMillis` of 23 milliseconds. The query had to examine every document in the collection and took 23ms even though `nReturned` was only 273 documents. While 23ms may not sound like much, this can get much longer for a collection housing one million documents.
```